gtda/homology/point_clouds.py

```diff
@@ -221,3 +227,233 @@ def transform(self, X, y=None):
         Xt = np.stack(Xt)
         Xt = np.nan_to_num(Xt, posinf=self.infinity_values_)
         return Xt
+
+
+@adapt_fit_transform_docs
+class SparseRipsPersistence(BaseEstimator, TransformerMixin):
+    """`Persistence diagrams <https://giotto.ai/theory>`_ resulting from
+    `Sparse Vietoris-Rips filtrations <https://giotto.ai/theory>`_.
+
+    Given a `point cloud <https://giotto.ai/theory>`_ in Euclidean space,
+    or an abstract `metric space <https://giotto.ai/theory>`_ encoded by a
+    distance matrix, information about the appearance and disappearance of
+    topological features (technically, `homology classes
+    <https://giotto.ai/theory>`_) of various dimensions and at different
+    scales is summarised in the corresponding persistence diagram.
+
+    Parameters
+    ----------
+    metric : string or callable, optional, default: ``'euclidean'``
+        If set to `'precomputed'`, input data is to be interpreted as a
+        collection of distance matrices. Otherwise, input data is to be
+        interpreted as a collection of point clouds (i.e. feature arrays),
+        and `metric` determines a rule with which to calculate distances
+        between pairs of instances (i.e. rows) in these arrays.
+        If `metric` is a string, it must be one of the options allowed by
+        :func:`scipy.spatial.distance.pdist` for its metric parameter, or a
+        metric listed in :obj:`sklearn.pairwise.PAIRWISE_DISTANCE_FUNCTIONS`,
+        including "euclidean", "manhattan", or "cosine".
+        If `metric` is a callable function, it is called on each pair of
+        instances and the resulting value recorded. The callable should take
+        two arrays from the entry in `X` as input, and return a value
+        indicating the distance between them.
+
+    homology_dimensions : iterable, optional, default: ``(0, 1)``
+        Dimensions (non-negative integers) of the topological features to be
+        detected.
+
+    coeff : int prime, optional, default: ``2``
+        Compute homology with coefficients in the prime field
+        :math:`\\mathbb{F}_p = \\{ 0, \\ldots, p - 1 \\}` where
+        :math:`p` equals `coeff`.
+
+    epsilon : float between 0. and 1., optional, default: ``0.1``
+        Parameter controlling the approximation to the exact Vietoris-Rips
```

Comment:
There was a rewording suggestion here which seems to have been ignored.

Reply:
But maybe I was wrong -- i.e. is this really approximating the _filtration_ and therefore the final persistent barcode as a result, or is the filtration the same but there is another clever way of approximating the persistence calculation>

Reply:
AFAIK, the filtration is different. See http://gudhi.gforge.inria.fr/doc/latest/class_gudhi_1_1rips__complex_1_1_sparse__rips__complex.html

Reply:
Agreed, thanks for clarifying!